server/app/repository/ExportServiceRepository.java

```diff
@@ -0,0 +1,93 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Provider;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import models.Version;
+import services.question.types.MultiOptionQuestionDefinition;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionType;
+
+/** Implements queries related to CSV exporting needs. */
+public final class ExportServiceRepository {
+  private final Database database;
+  private final Provider<VersionRepository> versionRepositoryProvider;
+
+  @Inject
+  public ExportServiceRepository(Provider<VersionRepository> versionRepositoryProvider) {
+    this.database = DB.getDefault();
+    this.versionRepositoryProvider = checkNotNull(versionRepositoryProvider);
+  }
+
+  /**
+   * This method creates a map of all the optionIds to its option's adminName string for a given
+   * multiselect question definition There are two queries which are run here 1. To get all the
+   * adminNames to optionIds for a given question 2. To get all the options which were ever picked
+   * by an applicant (this is to filter out options accidentally introduced) The options are now
+   * combined with the latest active version of the question and returned as an immutable map. It
+   * throws RuntimeException when the question is not of type Checkbox It also throws
+   * NoSuchElementException when the active version of the question is missing
+   */
+  public ImmutableMap<Long, String> getMultiSelectedHeaders(QuestionDefinition questionDefinition) {
+    if (!questionDefinition.getQuestionType().equals(QuestionType.CHECKBOX)) {
+      throw new RuntimeException("The Question Type is not checkbox");
+    }
+    String questionName = questionDefinition.getName();
+    Map<Long, String> alloptionsMap = new HashMap<>();
+    database
+        .sqlQuery(
+            "SELECT DISTINCT jsonb_array_elements(q.question_options)->>'adminName'AS"
+                + " AdminName,jsonb_array_elements(q.question_options)->>'id' AS Id FROM questions"
+                + " q where name = :currentQuestion::varchar")
+        .setParameter("currentQuestion", questionName)
+        .findList()
+        .stream()
+        .forEach(row -> alloptionsMap.put(row.getLong("id"), row.getString("AdminName")));
+    Set<Long> allSelectedOptions = new HashSet<>();
+    database
+        .sqlQuery(
+            "select json_array_elements(((object #>>"
+                + " '{}')::jsonb)::json#>'{applicant, "
+                + questionName
```

Comment:
**issue** and **suggestion**: We need to use the `setParameter()` strategy you used in the above query so we can guard against dependency injection, since questionName is ultimately a user-provided string.